### ISSUE TITLE
Closure route handler doesnt' support $this

### ISSUE THREAD

**Issue by joke2k** [PHP-DI/Slim-Bridge#52]
The Invoker doesn't bind the container with route closure.

```
use Psr\Http\Message\ResponseInterface as Response;
use Psr\Http\Message\ServerRequestInterface as Request;
use DI\Bridge\Slim\Bridge as AppFactory;

require __DIR__ . '/../vendor/autoload.php';

$app = AppFactory::create();

// ... $app->getContainer()->set('view', ... );

$app->get('/[{name}]', function (Request $request, Response $response, $name = null) {
    $str = $this->get('view')->fetchFromString(
        '<p>Hi, my name is {{ name }}.</p>',
        ['name' => $name ?? 'Anonymous']
    );
    $response->getBody()->write($str);
    return $response;
});

$app->run();
```

raises:

`Uncaught Error: Using $this when not in object context`

**Comment by mnapoli** [comment]
Are you sure about your example? `$this` is not used in your controller.

**Comment by joke2k** [comment]
wrong copy&paste sorry.
I edited with the right code.

**Comment by mnapoli** [comment]
OK I see what you mean, but this is expected behavior. With the PHP-DI bridge, dependencies are supposed to be passed via closure parameters.

I'm not sure if we should enable that behavior, if several people want this we may consider this.

**Comment by dakujem** [comment]
I believe it should be mentioned and _highlighted_ somewhere. I was just about to post this as an issue. 🤔

The point is, the Slim framework documetnation is explicit about this behaviour and one might be surprised to see it not working.

I actually find the original Slim's behaviour a **caveat**, but I was about to document it for my team and found it was no longer present. 😳
Then I spent quite a few minutes discovering what the difference was, as I did not see it mentioned anywhere in PHP-DI docs.

Something like this in the bridge docs, or on the website, would help:

>
> 💡
>
> PHP-DI will no longer bind `$this` and `static` to the container instance, all controller dependencies should be type-hinted instead.
>
